pkg/resourcemanager/controller/managedresource/merger.go

```diff
@@ -100,7 +100,7 @@ func merge(origin string, desired, current *unstructured.Unstructured, forceOver
 	if annotations[resourcesv1alpha1.PreserveReplicas] == "true" {
 		preserveReplicas = true
 	}
-	preserveResources := false
+	var preserveResources bool
 	if annotations[resourcesv1alpha1.PreserveResources] == "true" {
 		preserveResources = true
 	}
```

Comment:
This is the only change where I don't see the benefit.
According to the documentation of [QF1007](https://staticcheck.dev/docs/checks#QF1007) shouldn't it look like this?
```suggestion
	preserveResources := annotations[resourcesv1alpha1.PreserveResources] == "true"
```